Kube-controller-manager panic when wrong Azure VMSS name is configured

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:

Kube-controller-manager panic when wrong Azure VMSS name is configured:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2b70310]

goroutine 2786 [running]:
k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure.(*scaleSet).EnsureHostsInPool.func1(0x0, 0x1)
        /go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure/azure_vmss.go:1412 +0xe0
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/errors.AggregateGoroutines.func1(0xc001755c80, 0xc001dae000)
        /go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/errors/errors.go:217 +0x27
created by k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/errors.AggregateGoroutines
        /go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/errors/errors.go:217 +0x8d
```

This is because the error is not validated first in the following code path:

https://github.com/kubernetes/kubernetes/blob/2ded1e0c59d16aeef948dae5b92035292f213f3a/staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss.go#L1408-L1419

Azure cloud provider should check whether err happens and whether vmss is nil here.

**What you expected to happen**:

Azure cloud provider should report errors on such case, so that customers could get the root raw error and then fix the configuration.

**How to reproduce it (as minimally and precisely as possible)**:

Create Kuberenetes cluster with basic load balancer, configure a wrong VMSS name and then create/delete LoadBalancer typed services.

**Anything else we need to know?**:

The issue also happens on older versions, so the fixes need to cherry-pick on other branches.

/area provider/azure
/sig cloud-provider
/milestone v1.20

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

--- a/staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss.go
+++ b/staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss.go
@@ -1406,16 +1406,15 @@ func (ss *scaleSet) ensureBackendPoolDeletedFromVMSS(service *v1.Service, backen
 
 	for vmssName := range vmssNamesMap {
 		vmss, err := ss.getVMSS(vmssName, azcache.CacheReadTypeDefault)
+		if err != nil {
+			return err
+		}
 
 		// When vmss is being deleted, CreateOrUpdate API would report "the vmss is being deleted" error.
 		// Since it is being deleted, we shouldn't send more CreateOrUpdate requests for it.
 		if vmss.ProvisioningState != nil && strings.EqualFold(*vmss.ProvisioningState, virtualMachineScaleSetsDeallocating) {
 			klog.V(3).Infof("ensureVMSSInPool: found vmss %s being deleted, skipping", vmssName)
 			continue
-		}
-
-		if err != nil {
-			return err
 		}
 		if vmss.VirtualMachineProfile.NetworkProfile.NetworkInterfaceConfigurations == nil {
 			klog.V(4).Infof("EnsureHostInPool: cannot obtain the primary network interface configuration, of vmss %s", vmssName)

